### ISSUE TITLE
Insufficient DWARF-5 support: `AssertionError` on binaries from GCC 11

### ISSUE THREAD

**Issue by samuelsadok** [eliben/pyelftools#391]
pyelftools fails to parse certain files compiled with GCC 11 (sample: [main.zip](https://github.com/eliben/pyelftools/files/7853967/main.zip)), presumably due to the lack or incompleteness of DWARF-5 support:

```
>>> import elftools.elf.elffile
>>> elffile = elftools.elf.elffile.ELFFile(open('main.elf', 'rb'))
>>> cus = list(elffile.get_dwarf_info().iter_CUs())
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python3.10/site-packages/elftools/dwarf/dwarfinfo.py", line 353, in _parse_CUs_iter
    cu = self._cached_CU_at_offset(offset)
  File "/usr/lib/python3.10/site-packages/elftools/dwarf/dwarfinfo.py", line 383, in _cached_CU_at_offset
    cu = self._parse_CU_at_offset(offset)
  File "/usr/lib/python3.10/site-packages/elftools/dwarf/dwarfinfo.py", line 417, in _parse_CU_at_offset
    cu_structs = DWARFStructs(
  File "/usr/lib/python3.10/site-packages/elftools/dwarf/structs.py", line 92, in __init__
    assert address_size == 8 or address_size == 4
AssertionError
```
Adding debug output reveals that it's the third call to `DWARFStructs.__init__()` that crashes:
 - 1st and 2nd call:
    ```
    little_endian: True
    dwarf_format: 32
    address_size: 4
    dwarf_version: 2
    ```
 - 3rd call:
    ```
    little_endian: True
    dwarf_format: 32
    address_size: 0
    dwarf_version: 5
    ```
Therefore I assume that the builtin libraries of GCC 11 are compiled with DWARF-5 instead of DWARF-2 and that makes pyelftools crash. I see that DWARF-5 has been discussed in https://github.com/eliben/pyelftools/issues/325 but I'm not sure if my version of pyelftools is expected to support this. If this is expected, please close this issue and hopefully it will direct others to the root issue.

The sample binary was generated like this:
main.c:
```
int main() {
    return 0;
}
```
compile with:
```
arm-none-eabi-gcc main.c -g -specs=nosys.specs -o main.elf
```

 - `arm-none-eabi-gcc --version`: 11.2.0
 - `elftools.__version__`: '0.27'
 - `python --version`: Python 3.10.1

When repeating the same steps with GCC 10.3.0, pyelftools loads the resulting file correctly and does not throw an exception. Adding more aggressive optimization (`-flto`) also makes the binary load, but only as long as you don't use certain library features.

**Comment by sevaa** [comment]
The current master version of pyelftools opens the attached file and successfully scrolls through the DIEs of the one and only CU. The released version is several months behind. Maybe a release would be in order.

**Comment by woodruffw** [comment]
Yeah, a release should fix this -- I was also able to confirm that the DIEs parse correctly on `master`.

**Comment by eliben** [comment]
I've just released version 0.28

Closing the issue; please reopen if needed.
